Add option to auto-expand the filetree at startup

b0o opened the pull request:
`-autoexpandDepth` controls the depth to which the file tree should be expanded initially:

- `-1`: all levels should be expanded
- `0`: no levels should be expanded
- `1`: (default) only the root node should be expanded, as in the hitherto default behavior.
- `>1`: this many levels should be expanded, indexed from 1 starting at the root

`-autoexpandIgnore` allows directories to be excluded from auto-expanding if they match a regular expression. The expression is tested against a directory's path relative to the root.

wvanlint commented:
Sorry for the delay and thanks for the contribution!

I think `Traverse(false, ...)` does not limit the traversal in any way, so even if the auto-expand depth is small, it might still exhaustively traverse a large tree. Perhaps a better way would be to generalize the `Traverse` method to `Traverse(shouldTraverse func(*FileTree, int) bool, order Order, f func(*FileTree, int) error) error`, where the first predicate can limit the entire branch, for example:
- When nodes are not visible.
- When they match the ignore regex.
- When they exceed the maximum depth.

What do you think?

b0o commented:
Thanks for catching that! I looked into it and it turns out I misunderstood the behavior of the `visibleOnly` argument. Using `true` works and avoids this issue. I fixed that and also added some tests.

I like your idea of generalizing `Traverse`, and I actually implemented that before realizing the fix was simple. Since it's not necessary, I think it's fine to leave `Traverse` alone.

wvanlint commented:
Looks great, thanks for doing this!
